### DISCUSSION TITLE
[How to] Validate Foundry Course NFT using Opensea Testnet API [Sepolia Only]

### DISCUSSION THREAD

**Discussion by itznishant** [Cyfrin/foundry-full-course-cu#383] [General]
### [How to] Validate Foundry Course NFT using Opensea Testnet API [Sepolia Only]

**Validation URL (Opensea):**

`https://testnets-api.opensea.io/asset/sepolia/[Foundry-Course-NFT-Contract-Address]/[token-id]/validate
`

**Requirements:**

- Contract Address of Foundry Course NFT => 0x76B50696B8EFFCA6Ee6Da7F6471110F334536321

- NFT Token Id => e.g. 1045

**Steps:**

- Input these values in the _Opensea Validation URL_ and Hit Enter.

- There should be NO _errors_ in the validation output and the NFT _token_uri_ would be displayed as per below screenshot:

<img width="1179" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/17523443/2a31ac00-b136-4b0f-a66c-f0c2b78b17bf">

- You should copy this _token_uri_ and paste in a new browser tab to view the associated NFT metadata details (as below):

<img width="1403" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/17523443/9fd11994-1192-41cd-b073-46ad9fb66179">

- To view actual NFT image, select the  _image_ attribute value (starting with 'Qm') and paste this in another browser tab.

`https://ipfs.io/ipfs/[paste the CID value starting with 'Qm']
`

- For e.g.

`https://ipfs.io/ipfs/QmYNk9TaRp9QDRjDZYxrwh5xQGMeC9zY23tB4fs5qby6KT
`

- Now you can see the corresponding Foundry Course NFT Image!

<img width="1274" alt="Screenshot 2023-07-20 at 12 26 32 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/17523443/1fa89fce-c85b-4ad6-90f5-e018642864cf">

<-------------------------------------------------------------    END  ----------------------------------------------------------------->

This is my first contribution in this Repo / Discussions forums.

Do Give this Post a Like & Upvote!

Thanks to @PatrickAlphaC for this amazing Course.

Cheers,
**Nishant**

**Comment by PatrickAlphaC** [comment]
Awesome!!

I'd love to see a post on [medium](https://medium.com/), [mirror](https://mirror.xyz/), or [hashnode](https://hashnode.com/) on this!

**Reply by itznishant** [reply]
Published on Mirror!

https://mirror.xyz/0x6C28Ad0CaCEAc64C8a15535E6744B6847EA2AB0B/V9f3mI4Jj_5Bt7unKQoglUF_IXo-6dXuCVc9O6BN4bc

@PatrickAlphaC thanks for the suggestion.

**Reply by itznishant** [reply]
I went ahead and created an explainer video on this using AI.

More details in [this](https://github.com/Cyfrin/foundry-full-course-f23/discussions/430) discussion in "Show and tell".

@PatrickAlphaC

**Comment by itznishant** [comment]
Hopefully, the article on Mirror.xyz is visible to all, as it seems to show as draft still.  🤔

**Reply by itznishant** [reply]
This is resolved now. The article is published at mirror link above.

**Reply by PatrickAlphaC** [reply]
awesome!!
